From c56341f24f719f2dafdb2d96a0698b64c810d209 Mon Sep 17 00:00:00 2001
From: ismum chowdhury <ichowdhury4@fordham.edu>
Date: Thu, 1 Apr 2021 13:29:51 -0400
Subject: [PATCH] create patch file

---
 client.py | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/client.py b/client.py
index 40badcd..47199ec 100644
--- a/client.py
+++ b/client.py
@@ -35,14 +35,13 @@ def getDataPoint(quote):
 	stock = quote['stock']
 	bid_price = float(quote['top_bid']['price'])
 	ask_price = float(quote['top_ask']['price'])
-	price = bid_price
+	price = (bid_price + ask_price)/2 
 	return stock, bid_price, ask_price, price
 
 def getRatio(price_a, price_b):
-	""" Get ratio of price_a and price_b """
-	""" ------------- Update this function ------------- """
-	""" Also create some unit tests for this function in client_test.py """
-	return 1
+	if(price_b == 0):
+		return
+	return  price_a/price_b
 
 # Main
 if __name__ == "__main__":
@@ -50,10 +49,10 @@ if __name__ == "__main__":
 	# Query the price once every N seconds.
 	for _ in xrange(N):
 		quotes = json.loads(urllib2.urlopen(QUERY.format(random.random())).read())
-
-		""" ----------- Update to get the ratio --------------- """
+		prices = {}
 		for quote in quotes:
 			stock, bid_price, ask_price, price = getDataPoint(quote)
-			print "Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price)
+			prices[stock] = price
+			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-		print "Ratio %s" % getRatio(price, price)
+		print ("Ratio %s" % getRatio(prices['ABC'], prices['DEF']))
-- 
2.22.0

